logstash-core/lib/logstash/runner.rb

```diff
@@ -267,7 +267,7 @@ class LogStash::Runner < Clamp::StrictCommand
          I18n.t("logstash.runner.flag.event_api.tags.illegal"),
          :default => LogStash::SETTINGS.get_default("event_api.tags.illegal"),
          :attribute_name => "event_api.tags.illegal", :passthrough => true,
-         :deprecated_msg => I18n.t("logstash.runner.tags-illegal-deprecated")
+         :obsoleted_version => "9"
```

Comment:
The reason for the refactoring is, came with an idea to widely use `obsoleted_version` with other deprecation params (such as `http.*`). Change LGTM!

One thing I would like to clarify: merging against 8.x would make sense since `event_api.tags.illegal` will be removed from `main` (with [this PR](https://github.com/elastic/logstash/pull/16461)), how do you think?

Reply:
I need this PR to merge to main because the way of handling deprecation warning in `clamp.rb` is benefit for the future deprecation not limit to 8.x